### PR TITLE
Part 1

### PR DESCRIPTION
**Pull request by Lauren34**

### CODE REVIEW

File: src/main/java/lesser/citibike/CitiBikeHelper.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package lesser.citibike;
+
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CitiBikeHelper {
```
Comment (andrewoid) [inline src/main/java/lesser/citibike/CitiBikeHelper.java:8]: This class doesn't actually help. You are making the network calls multiple times when you should only be making the calls once.

**Comment by andrewoid** [comment]
The .idea folder should not be added to git. You need to remove those files and then add it to your gitignore

File: src/test/java/lesser/citibike/CitiBikeHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        // when
+        StationsResponse.Station closestStation = helper.findClosestStationWithDocks(lat, lon).blockingGet();
+
+        // then
+        assertNotNull(closestStation, "Station should not be null");
+        assertTrue(closestStation.lat > 0, "Latitude should be valid");
+        assertTrue(closestStation.lon < 0, "Longitude should be valid");
+    }
+
+    private double haversine(double lat1, double lon1, double lat2, double lon2) {
```
Comment (andrewoid) [inline src/test/java/lesser/citibike/CitiBikeHelperTest.java]: Why is this here?

Comment (Lauren34) [reply src/test/java/lesser/citibike/CitiBikeHelperTest.java]: Got rid of it, wasn't supposed to be there

File: src/main/java/lesser/citibike/CitiBikeHelper.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    public CitiBikeHelper(CitiBikeService service) {
+        this.service = service;
+    }
+
+    private Single<StationsResponse> getStations() {
+        if (stationsResponse == null) {
+            stationsResponse = service.getStations().subscribeOn(Schedulers.io());
+        }
+        return stationsResponse;
+    }
```
Comment (andrewoid) [inline src/main/java/lesser/citibike/CitiBikeHelper.java]: This doesn't work the way you think it does if you call it twice. 

I'm impressed that you're trying to use operators like zip and map, but this is not the way that I showed you how to use retrofit.

File: src/test/java/lesser/citibike/CitiBikeHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        String stationId = "816e50eb-dc4b-47dc-b773-154e2020cb0d";
+
+        // when
+        StatusResponse.StationStatus status = null;
+        for (StatusResponse.StationStatus s : statusResponse.data.stations) {
+            if (s.station_id.equals(stationId)) {
+                status = s;
+                break;
+            }
+        }
```
Comment (andrewoid) [inline src/test/java/lesser/citibike/CitiBikeHelperTest.java]: This should be a method in your `helper`

**Review by andrewoid — APPROVED** [review]
